sys/libc/string.c

```diff
@@ -50,4 +52,30 @@ const void *memchk(const void *data, uint8_t c, size_t len)
     return NULL;
 }
 
+int swprintf(string_writer_t *sw, const char *restrict format, ...)
+{
+    va_list args;
+    int res;
+
+    /* make sure flash_vsnprintf() is not used */
+    #if HAS_FLASH_UTILS_ARCH
+    #undef vsnprintf
+    #endif
+
+    va_start(args, format);
+    res = vsnprintf(sw->position, sw->capacity, format, args);
+    va_end(args);
+
+    if (res <= (int)sw->capacity) {
```

Comment:
if `res == capacity`, the output was truncated by one character. This check should therefore be `<`.